### PR TITLE
Composite layer interface and props transfer

### PR DESCRIPTION
**Pull request by shaojingli**
Change the interface of GeoJson layer and Polygon layer to clear the confusion
Change the way composite layers pass their props to their underlying layers. Now it requires explicitly passing. Layers affected are the GeoJsonLayer, PolygonLayer, GridLayer, and HexagonLayer.

Docs change will be in an updated #436 

@Pessimistress @ibgreen @gnavvy

### CODE REVIEW

File: src/layers/core/geojson-layer/geojson-layer.js [hunk 1]
```diff
@@ -80,13 +84,15 @@ export default class GeoJsonLayer extends CompositeLayer {
...
+      getLineWidth, getElevation, updateTriggers} = this.props;
     const {id, stroked, filled, extruded, wireframe} = this.props;
+    const {lineWidthScale, lineWidthMinPixels, lineWidthMaxPixels,
+      lineJointRounded, lineMiterLimit, fp64} = this.props;
 
-    let {} = this.props;
     const drawPoints = pointFeatures && pointFeatures.length > 0;
     const drawLines = lineFeatures && lineFeatures.length > 0;
-    const hasPolygonOutline = polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
+    const hasPolygonLine = polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
```
Comment (ibgreen) [inline src/layers/core/geojson-layer/geojson-layer.js]: hasPolygonLines?

File: src/layers/core/geojson-layer/geojson-layer.js [hunk 2]
```diff
@@ -95,91 +101,106 @@ export default class GeoJsonLayer extends CompositeLayer {
...
-        onHover,
-        onClick
-      }));
+        }
+      });
 
     const polygonWireframeLayer = wireframe &&
       extruded &&
       hasPolygon &&
-      new SolidPolygonLayer(Object.assign({}, this.props, {
```
Comment (ibgreen) [inline src/layers/core/geojson-layer/geojson-layer.js]: Looks good to get rid of the `Object.assign` here.

File: src/layers/core/geojson-layer/geojson-layer.js [hunk 3]
```diff
@@ -95,91 +101,106 @@ export default class GeoJsonLayer extends CompositeLayer {
...
 
     return [
       polygonFillLayer,
       polygonWireframeLayer,
-      polygonOutlineLayer,
-      lineLayer,
+      polygonLineLayer,
+      pathLayer,
       pointLayer
     ].filter(Boolean);
```
Comment (ibgreen) [inline src/layers/core/geojson-layer/geojson-layer.js]: We should to add the filtering by default inside deck.gl - Because I see so many layers doing `.filter(Boolean)` thing now. React supports `null`/empty elements, we should too. I think it is a one line change, could even go into this PR.

File: src/layers/core/grid-layer/grid-layer.js [hunk 4]
```diff
@@ -35,6 +35,7 @@ const defaultProps = {
   elevationRange: defaultElevationRange,
   elevationScale: defaultElevationScale,
   getPosition: x => x.position,
+  extruded: false,
   fp64: false
   // AUDIT - getWeight ?
```
Comment (ibgreen) [inline src/layers/core/grid-layer/grid-layer.js]: Remove?

File: src/layers/core/geojson-layer/geojson-layer.js (hunk 3 above)
Comment (shaojingli) [inline src/layers/core/geojson-layer/geojson-layer.js]: Agree. but let's do it in a different PR
